### PR TITLE
Handle cache start finalize synchronously

### PR DESCRIPTION
**Pull request by robertbaldyga**
Signed-off-by: Robert Baldyga <robert.baldyga@intel.com>

### CODE REVIEW

**Review by arutk — CHANGES_REQUESTED** [review]

File: modules/cas_cache/layer_cache_management.c [hunk 1]
```diff
@@ -1834,72 +1834,26 @@ static void init_instance_complete(struct _cache_mngt_attach_context *ctx,
...
 	struct _cache_mngt_attach_context *ctx = priv;
-	struct cache_priv *cache_priv = ocf_cache_get_priv(cache);
 	int caller_status = _cache_mngt_async_callee_peek_result(&ctx->async);
 
-	if (caller_status || error) {
-		if (error == -OCF_ERR_NO_FREE_RAM && ctx->cmd) {
-			ocf_mngt_get_ram_needed(cache, ctx->device_cfg,
-					&ctx->cmd->min_free_ram);
-		} else if (caller_status == -KCAS_ERR_WAITING_INTERRUPTED) {
+	if (error == -OCF_ERR_NO_FREE_RAM && ctx->cmd) {
```
[the remark this answers is not in the document]
Comment (robertbaldyga) [reply modules/cas_cache/layer_cache_management.c:1842]: I think it doesn't hurt to inform user about that even if operation was interrupted.

File: modules/cas_cache/layer_cache_management.c [hunk 2]
```diff
@@ -1834,72 +1834,26 @@ static void init_instance_complete(struct _cache_mngt_attach_context *ctx,
...
-	kthread_stop(rollback_thread);
-
-	ocf_mngt_cache_unlock(cache);
-}
-
 static void _cache_mngt_start_complete(ocf_cache_t cache, void *priv, int error)
 {
 	struct _cache_mngt_attach_context *ctx = priv;
-	struct cache_priv *cache_priv = ocf_cache_get_priv(cache);
 	int caller_status = _cache_mngt_async_callee_peek_result(&ctx->async);
```
[the remark this answers is not in the document]
Comment (robertbaldyga) [reply modules/cas_cache/layer_cache_management.c]: Done.

**Review by arutk — APPROVED** [review]

**Comment by robertbaldyga** [comment]
Fixes #609

**Comment by robertbaldyga** [comment]
retest this please

**Pull request merged**